build-images.sh

```diff
@@ -9,50 +9,63 @@ function usage() {
     exit 0
 }
 
-while [ -n "$1" ]; do
-    case $1 in
-    -h | --help) usage ;;
-    -?*)
-        echo "Unknown option '$1'" 1>&2
-        exit 1
+parse_args() {
+  local SHORT=h,d:
+  local LONG=distro:
+
+  local OPTS=$(getopt -o $SHORT -l $LONG --name "$0" -- "$@")
+
+  if [ $? != 0 ] ; then echo "Failed to parse options...exiting." >&2 ; exit 1 ; fi
+
+  eval set -- "$OPTS"
+
+  . settings.sh  # set initial values
+
+  while true ; do
+    case "$1" in
+      -d | --distro )
+        ROS_DISTRO="$2"
+        shift 2
         ;;
-    *)
+      --test )
+        BUILD_TEST="$2"
+        shift 2
+        ;;
+      -- )
+        shift
         break
         ;;
+      -h | *)
+        usage
+        shift
+        ;;
     esac
-    shift
-done
-
-ROS_DISTRO=${1-noetic}
-ROS_VERSION=""
-UBUNTU_DISTRO=""
-
-case $ROS_DISTRO in
-noetic)
-    ROS_VERSION=""
-    UBUNTU_DISTRO="focal"
-    ;;
-melodic)
-    ROS_VERSION=""
-    UBUNTU_DISTRO="bionic"
-    ;;
-kinetic)
-    ROS_VERSION=""
-    UBUNTU_DISTRO="xenial"
-    ;;
-foxy)
-    ROS_VERSION=2
-    UBUNTU_DISTRO="focal"
-    ;;
-dashing)
-    ROS_VERSION=2
-    UBUNTU_DISTRO="bionic"
-    ;;
-*)
-    echo "Not supported ROS_DISTRO '$ROS_DISTRO'" 1>&2
+  done
+
+}
+
+validate_ros_distro() {
+  local distro=$1
+  if [[ ! $VALID_ROS_DISTROS =~ (^|[[:space:]])"${distro}"($|[[:space:]]) ]]; then
+    echo "Not supported ROS_DISTRO '${distro}'" &2
+    usage
     exit 1
-    ;;
-esac
+  fi
+}
+
+build_image() {
+  validate_ros_distro ${ROS_DISTRO}
+
+  if [[ "$BUILD_TEST" == "true" ]]; then
+    docker build --build-arg ROS_DISTRO=$ROS_DISTRO -t orise-robotics/ros-$ROS_DISTRO:test -f Dockerfile.test .
+  fi
+
+  docker build --build-arg ROS_DISTRO=$ROS_DISTRO -t orise-robotics/ros-$ROS_DISTRO:devel .
```

Comment:
I think we can change this behavior to build only the `devel` or the `test` images depending on the `--test` argument

Reply:
changed